tools/eggnog_mapper/eggnog_mapper.xml

```diff
@@ -0,0 +1,491 @@
+            <param name="eggnog_data" type="select" label="Version of eggNOG Database">
+                <options from_data_table="eggnog_mapper">
+                    <filter type="static_value" name="type" value="data" column="2"/>
+                </options>
+            </param>
+            <param name="translate" type="boolean" truevalue="--translate" falsevalue="" checked="false"
+                label="Are these coding DNA sequences that need to be translated?"/>
+            <conditional name="db">
+            <param name="mode" type="select" label="Annotation Type">
+                <option value="hmmer">HMM</option>
+                <option value="diamond">DIAMOND</option>
+            </param>
+            <when value="hmmer">
+                <param name="database" type="select" label="HMM target database" help="Choose either the full eggNOG database or a subset of it. Contact your Galaxy admin to have additional HMM databases installed.">
+                    <options from_data_table="eggnog_mapper">
+                        <filter type="static_value" name="type" value="hmmer" column="2"/>
+                        <validator type="no_options" message="No HMM databases are available; request installation from your Galaxy admin." />
+                    </options>
+                </param>
+                <conditional name="hmm_options">
+                        <param name="hmm_settings" type="select" label="HMM Search Options">
+                            <option value="default">Use defaults</option>
+                            <option value="specified">Set HMM Search Options</option>
+                        </param>
+                        <when value="default"/>
+                        <when value="specified">
+                            <param name="hmm_maxhits" type="integer" value="1" label="Max number of hits to report per query sequence"/>
+                            <param name="hmm_evalue" type="float" min="0" value="0.001" label="E-value threshold" />
+                            <param name="hmm_score" type="integer" min="0" value="20" label="Bit score threshold" />
+                            <param name="hmm_maxseqlen" type="integer" value="5000" label="Ignore query sequences larger than `maxseqlen`.  Default=5000" />
+                            <param name="hmm_qcov" type="float" value="" min="0." max="1." optional="true" label="min query coverage (from 0 to 1). Default=(disabled)" />
+                            <param name="Z" type="integer" value="40000000" min="0" label="Fixed database size used in phmmer/hmmscan (allows comparing e-values among databases).  Default=40,000,000" />
+                        </when>
+                </conditional>
+                </when>
+                <when value="diamond">
+                    <!-- db.database is used in diamond mode only to name outputs -->
+                    <param name="database" type="hidden" value="diamond"/>
```

Comment:
why is this needed?

Reply:
this is used to automatically name the outputs in 

`${tool.name} on ${on_string}: ${db.database}.seed_orthologs`,

because diamond mode doesn't require a database name to be passed